gateway/app/main/codes.py

```diff
@@ -1,18 +1,44 @@
-class GRID_MSG(object):
-    TYPE_FIELD = "type"
-    DATA_FIELD = "data"
+class MSG_FIELD:
+    TYPE = "type"
+    DATA = "data"
+    WORKER_ID = "worker_id"
+    MODEL = "model"
+    ALIVE = "alive"
+
+
+class CONTROL_EVENTS(object):
     SOCKET_PING = "socket-ping"
-    GET_PROTOCOL = "get-protocol"
+
+
+class WEBRTC_EVENTS(object):
     PEER_LEFT = "webrtc: peer-left"
     INTERNAL_MSG = "webrtc: internal-message"
     JOIN_ROOM = "webrtc: join-room"
 
 
+class FL_EVENTS(object):
+    REPORT = "federated/report"
+    AUTHENTICATE = "federated/authenticate"
+    CYCLE_REQUEST = "federated/cycle-request"
+
+
+class CYCLE(object):
+    STATUS = "status"
+    KEY = "request_key"
+    PING = "ping"
+    DOWNLOAD = "download"
+    UPLOAD = "upload"
+    VERSION = "version"
+    PLANS = "plans"
+    PROTOCOLS = "protocols"
+    CLIENT_CONFIG = "client_config"
+    SERVER_CONFIG = "server_config"
+    TIMEOUT = "timeout"
+    DIFF = "diff"
+    AVG_PLAN = "averaging_plan"
+    SECURE_AGG_PROTOCOL = "secure_agg_protocol"
+    TRAINING_PLANS = "training_plans"
```

Comment:
"Training plans" is no longer accurate, as a plan could be used entirely just for inference or validation purposes. It's better to just call this "plans" or "client plans". Think of it this way:

**Client**
- Plan(s) (ONE or more)
- Protocol(s) (ZERO or more)
- Configuration (only one, just a Python dictionary)

**Server**
- Averaging plan (only one)
- Configuration (only one, just a Python dictionary)

**Used by client AND server**
- Model (only one)

Reply:
Done! :)